Feature/textanalytics5.2

suhas92 opened the pull request:
This PR adds 3.2-preview.2 endpoint SDK.

Things to look out for -

Swagger file - (swagger-latest.json) I have made few edits on the swagger to make sure it is autogenerating code for wrapper implementation

Auto - Generated Code - All the files under /Generated folder are coming from swagger file, if you think we should have a change in generated ones, please comment on the swagger-latest.json for more clarity.

For each customization, I have added a new class with the same name which will add the customization.

AnalyzeHealthOperation.cs - This is the class for implementing the LRO pattern as described by the Azure .NET SDK guidelines.
AnalyzeHealthResultCollection.cs - Added this class as a wrapper for ReadOnlyCOllection on top of HealthcareResult.cs
Some customizations are around making Models internal to the SDK and are not exposed to the customer.
Some customizations are for renaming classes as many were not showing a correct naming pattern.

suhas92 committed:
added 3.2-preview.1

WIP - updated swagger

WIP - build passing

AnalyzeHealthOperation wrapper working

added analyze operation wrapper

added tests  for analyze and analyzehealth

fix

suhas92 committed:
Added working sample for health async

christothes committed:
Add instructions to README for Managed HSM (#15720)

closes #15068 , #15306

MiYanni committed:
[ServiceBus] Changelog updates (#15731)

* Added changelog updates for release.

* Removed fixed changelog entries.

* Wrong class name was used.

* Fixed other class name.

azure-sdk committed:
Increment package version after release of Azure.Data.Tables (#15730)

pakrym committed:
Fix a bug with sample tests hitting recording guardrails (#15729)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/15728

maririos committed:
[TA] Update release date in changelog (#15732)

Set release day

jsquire committed:
[Event Hubs Client] Processor Options Fix (#15725)

The focus of these changes is to correct an issue where the load balancing
interval specified in the processing options was not respected and to add
the intended timing options to the `EventProcessorClientOptions`, which,
despite being part of the approved design, were overlooked and not implemented
as part of the load balancing strategy changes.

audunn committed:
[NetAppFiles] Anf 6288 net app files 2020 06 01 (#15643)

* Bump to api 2020-06-01, adding tests

* Update api and version numbers

* Update sesssion records

* Rename BackupPolicy test due to gitignore rule

* Remove extra dash

Co-authored-by: Audunn Baldvinsson <audunn.baldvinsson@netapp.com>

JoshLove-msft committed:
Update changelog for release (#15740)

rajkumar-rangaraj committed:
Azure Monitor Exporter - Add retry status code to service client (#15659)

* Add exception to track call

* Fix CI tests

areddish committed:
Regenerate Computer Vision SDK for v3.1 API (#15531)

* Regenerate Computer Vision SDK for v3.1 API

* Update and add unit test run captures

* Updates

* Remove extra / in Uri

shmartel committed:
Add support for new BYOK parameter keyVaultKeyUri (#15684)

* Add support for new BYOK parameter keyVaultKeyUri

* Since the parameter re-ordering in the ServicesProperties ctor is a breaking change, I've bumped the package version to 2.0.0

amnguye committed:
QueueServiceClient Properties incorrectly creating XML (Cors) (#15686)

kinelski committed:
[MetricsAdvisor] Adding documentation across the client: second batch (#15721)

Commit:
Azure Monitor Exporter - Split Transmitter into two classes (#15748)

* split transmitter into two classes

* cleanup

pakrym committed:
Add serializer support to JsonPatchDocument (#15754)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/15752

heaths committed:
Update Key Vault CHANGELOGs (#15757)

navysingla committed:
Adding reseller Id parameter (#15658)

azure-sdk committed:
Increment version for keyvault releases (#15760)

* Increment package version after release of Azure.Security.KeyVault.Keys

* Increment package version after release of Azure.Security.KeyVault.Certificates

* Increment package version after release of Azure.Security.KeyVault.Secrets

* Increment package version after release of Azure.Security.KeyVault.Administration

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1052 (#15668)

* Update location of patch files

* Add some changes in eng/common to test things

* Switch to pushing to upstream

* Change workflow to use upstream branches

Co-authored-by: Chidozie Ononiwu <chononiw@microsoft.com>

azure-sdk committed:
Increment package version after release of Azure.Messaging.ServiceBus (#15755)

azure-sdk committed:
Increment package version after release of Azure.Messaging.EventGrid (#15756)

azure-sdk committed:
Increment package version after release of Azure.AI.TextAnalytics (#15751)

pakrym committed:
Support regions without names in Update-Snippets (#15746)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/15744

azure-sdk committed:
Update AutoRest C# version to 1.0.0-alpha.20201006.1 (#15742)

danikaZ committed:
Add private endpoint name to net sdk (#13923)

timtay-microsoft committed:
Initial GA work for digital twins (#15734)

* Regenerate digital twins code for GA release

GA service swagger comes with new optional parameters for all service requests. Thankfully the swagger also conveniently groups them together with existing options so that we can directly use the generated options objects in our CL APIs (except for a few cases)

* Re-record tests

* Add ObjectSerializer, rename vars from board review, misc code clean up (#15753)

Co-authored-by: David R. Williamson <drwill@microsoft.com>

christothes committed:
[MetricsAdvisor] Updates for API consistency with Java (#15766)

[MetricsAdvisor] Updates for API consistency with Java
Co-authored-by: Caio Saldanha <camaiaor@microsoft.com>

seanmcc-msft committed:
Fixed xml comments for BlobServiceClient.FindBlobsByTag() (#15741)

juancamilor committed:
Updates in tests and remove client option param on GetChatThreadClient (#15711)

* - Fixed GetChatThreadClient to use client options from parent.

- Refactored tests to have :
-- Samples : Just used for snippets
--Recorded: For code coverage. Moved ReadReceipt assertions to own test to run in playback mode

* Minor changes on readme text

* Included re-recorded tests

* Minor wording updates

Co-authored-by: Camilo Ramirez <juramir@microsoft.com>

kasobol-msft committed:
[Storage][Webjobs] Split and hide storage account. decouple queues from blobs. (#15758)

* add webjobs account to arm.

* temporarily scope tests to webjobs only.

* Revert "temporarily scope tests to webjobs only."

This reverts commit e74a638b637855f6a99317bda4a03694800a5aa7.

* temporarlily scope webjobs tests attempt 2

* get rid of custom liveonly attribute.

* webjobs env vars should use connection strings.

* fix syntax.

* another syntax fix.

* does this help?

* does this help???

* fix more tests.

* Revert "temporarlily scope webjobs tests attempt 2"

This reverts commit 940a5a689a9c90fef162de5b6e31646daae9b215.

* pr feedback.

* pr feedback.

* remove shared loadbalancer queue from queues.

* replacement ?

* this compiles.

* queues are green.

* blobs green!

* all green

* this should help

* fix build.

* dropping some dependencies.

* untangle more dependencies.

* remove comment.

* pr feedback.

HarveyLink committed:
Update EventHub tests and rescords. (#15518)

Commit:
fix script Prepare-Release (#15772)

pakrym committed:
Update release date for OpenTelemetry (#15774)

seanmcc-msft committed:
Open Read Seekability (#15032)

Commit:
[Identity] updating changelog for 1.3.0-beta.2 release (#15780)

jsquire committed:
[Event Hubs Client] Disallow Cancellation after Publish (#15745)

The focus of these changes is to remove a cancellation token check at an
inappropriate location; if the send operation completes, cancellation
should not be honored.  Either the service exception or a success should
be returned so that callers clearly understand that events were published.

azure-sdk committed:
Increment package version after release of Azure.Identity (#15781)

pakrym committed:
Add methods to create Pageables from a list of pages (#15776)

pakrym committed:
Add support to enabling log forwarding explicitly and use it in storage. (#15767)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1066 (#15771)

* Update cache-control for newly created blobs

* remove extra

* remove extra space

* Added the cache control for existing blobs as well

* Update copy-docs-to-blobstorage.ps1

Co-authored-by: Sima Zhu <sizhu@microsoft.com>
Co-authored-by: Sima Zhu <48036328+sima-zhu@users.noreply.github.com>

kasobol-msft committed:
[Storage][Webjobs] Hide parameter descriptors (#15783)

* trimming.

* api

* poke CI

pakrym committed:
Use test environment credentials in extensions tests (#15723)

minnieliu committed:
Update README Nuget version (#15785)

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

heaths committed:
Fix Key Vault Administration README (#15762)

paterasMSFT committed:
Fixes integration tests hanging on CreateIfNotExistsAsync and DeleteIfExistsAsync (#15719)

* Fixes a few integration tests.

* Cleanup.

* More cleanup.

* More cleanup.

* Fixed a test broken in the previous PR.

* Made test synchronous.

* Fixes file access problem.

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1077 (#15768)

* Simplify Net-TestResources usage

* docs and windows check

* Update eng/common/TestResources/New-TestResources.ps1

Co-authored-by: Heath Stewart <heaths@outlook.com>

* update markdown

* make service directory the default parameter

* Fix links

* Doc change

Co-authored-by: Pavel Krymets <pavel@krymets.com>
Co-authored-by: Heath Stewart <heaths@outlook.com>

minnieliu committed:
Adding reference links to the nuget package descriptions (#15779)

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

kasobol-msft committed:
Hide blob name validation attribute. (#15794)

dochung4 committed:
Remove upper limit for DedicatedHostGroup's PlatformFaultDomainCount and DedicatedHost's PlatformFaultDomain (#15377)

* Updated

* Addressed comments

lululilliancoding committed:
Dev/lillian/set dot net sdk v20200202 preview short term retention policy 2 (#15713)

* Draft V2020-02-02 ShortTermRetention (STR) on SQL DB.
1) Unit testst not passed (Backend MS changes not finished yet). TODO:Re-run tests and re-copy test session.
2) STR API swagger in feature branch rather than master. TODO: Re-generate clients.

* regenerate clients via swagger master branch.

* remove non-related changes

* tiny change on str scenario tests

* update STR scenario tests - Passed on cluster Dogfood, south east asia

* re-upload passed live test record

Co-authored-by: Lillian Liu <liulu@microsoft.com>

kinelski committed:
[MetricsAdvisor] Added README and samples (#15793)

MiYanni committed:
[ServiceBus] Use BinaryData for G/SetSessionState. (#15656)

* [DOES NOT BUILD] Changed type from byte[] to BinaryData.

* Regen API. Update samples. Fix broken tests and samples.

* Fixed passing encoded strings instead of the strings directly to BinaryData.

* Pushed BinaryData into the internal transport layer.

* Fixed broken test.

kinelski committed:
Updated links to master (#15802)

kasobol-msft committed:
[Storage][Webjobs] Remove IWatcher related code (deprecated functionality). (#15799)

In this PR:
- Remove infrastructure related to "watch" parameters for dashboarding (deprecated and discontinued interfaces).
- Add missing coverage when blob gets written to (which matches V12 open write behavior). In V12 either action of commits a blob:
    - Flush (can commit multiple times, i.e. new block list will be committed and visible after that action).
    - Close/Dispose.

christothes committed:
[Metrics Advisor] Add remaining docstrings (#15792)

timtay-microsoft committed:
Update digital twins versions and changelog in preparation for GA release (#15775)

kinelski committed:
[MetricsAdvisor] Adding more docstrings (#15813)

christothes committed:
Add hooks doc strings (#15811)

kinelski committed:
[MetricsAdvisor] Final docstrings for beta 1 (#15816)

kinelski committed:
[MetricsAdvisor] Constructor bug fix (#15817)

kinelski committed:
[MetricsAdvisor] Fixed readme links (#15819)

kinelski committed:
[MetricsAdvisor] Update api view (#15820)

azure-sdk committed:
Increment version for metricsadvisor releases (#15821)

Increment package version after release of Azure.AI.MetricsAdvisor

maririos committed:
[FR] Rename DisplayName -> ModelName (#15815)

* add missing things

* rename displayname -> modelname

maririos commented:
/azp run net - textanalytics - ci

pakrym committed:
Add support for Is New Library field (#15814)

chidozieononiwu commented:
/azp where

chidozieononiwu commented:
/azp list

chidozieononiwu commented:
/azp run net - textanalytics - ci

pakrym committed:
Move configuration-based client logic to Microsoft.Azure.WebJobs.Extensions.Clients (#15804)

heaths committed:
Add READMEs for Microsoft.Spatial support packages (#15808)

* Add READMEs for Microsoft.Spatial support packages

* Resolve PR feedback

* Update CHANGELOGs for preview release

* Rename preview to beta

chidozieononiwu commented:
/azp run net - textanalytics - ci

maririos commented on sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeOperation.cs:
```diff
@@ -0,0 +1,211 @@
+
+                try
+                {
+                    bool? _showStats = null;
```
`_` prefix means the property is private. Because we want this library to follow .NET principles too, we follow the .NET coding styles => https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md

maririos replied:
also, why are all these null? shouldn't the customer be able to set this?

maririos commented on sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthResultCollection.cs:
```diff
@@ -0,0 +1,48 @@
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// Collection of <see cref="DetectLanguageResult"/> objects corresponding
```
DetectLanguageResult => HealthcareResult

maririos commented on sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthResultCollection.cs:
```diff
@@ -0,0 +1,48 @@
+        }
+
+        /// <summary>
+        /// Gets statistics about the documents batch and how it was processed
```
wrong docstring

maririos commented on sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthResultCollection.cs:
```diff
@@ -0,0 +1,48 @@
+        /// by the service.  This property will have a value when IncludeStatistics
+        /// is set to true in the client call.
+        /// </summary>
+        public IList<HealthcareResult> Documents { get; }
```
This shouldn't be a property as the list should be passed to the ReadOnlyCollection

suhas92 committed:
update health

maririos commented:
I think it will be better to keep the feature branch up to date with master and independently do this PR with your changes. Otherwise is going to be too hard to review.
Let me know if this is not clear

azure-sdk committed:
Support env vars with hyphens when printing output (#15818)

Co-authored-by: JoshLove-msft <54595583+JoshLove-msft@users.noreply.github.com>

tg-msft committed:
Search Fancy Batching (#15750)

Adds SearchIndexingBufferedSender to index search documents with intelligent
batching, automatic flushing, and retries for failed indexing actions.

Fixes #11161.

paterasMSFT committed:
Flaky test band-aid (#15838)

* cleanup

* Mark and (unreliably) mitigate flaky tests.

kasobol-msft committed:
Drop depedency on webjobs internals in testcommon. (#15835)

suhas92 committed:
pull in master

suhas92 commented:
> I think it will be better to keep the feature branch up to date with master and independently do this PR with your changes. Otherwise is going to be too hard to review.
> Let me know if this is not clear

@maririos - I do not have access to push to Azure, so I can't update the feature branch.

suhas92 committed:
WIP - analyze health care

amnguye committed:
Adding move, execute, list to blob sas permissions; Added ability to … (#16352)

* Adding move, execute, list to blob sas permissions; Added ability to set saoid and scid in blob sas; Fixed bug with using directory sas and shared key

nisha-bhatia committed:
Send SessionRecords to correct location (#15937)

* send SessionRecords to correct location

* add more error handling

* WIP: added dataplane and Track 2 correct locations for Session Records, remove Track 1, add isMgmtProject attribute to track 2 rps

* Update Directory.Build.props

* Update RecordedTestBase.cs

* add IsTestProject

* remove changes to eventhub

* remove unnecessary imports

* update docs and remove UpdateSessionRecords target

* update documentation and targets

* forgot to remove dotnet build

* Delete UpdateSessionRecords.ps1

Co-authored-by: bquantump <53361486+bquantump@users.noreply.github.com>

suhas92 committed:
WIP - stable build

Commit:
ADT: Change return type for GetRelationships to Pageable of T (#16353)

suhas92 committed:
Merge branch 'master' of https://github.com/Azure/azure-sdk-for-net into feature/textanalytics5.2
